### DISCUSSION TITLE
What is the best setting for minimum latency and a single final transcript for a deepgram real-time api for conversational bot

### DISCUSSION THREAD

**Discussion by karanbangia** [deepgram/community#588] [General help] [answered]
i want to achieve the minimum latency for real-time transcription, can you pls suggest best settings for the same, 
i am trying to play with the setting, pls help me how to improve this? 
{
      encoding: "mulaw",
      sample_rate: "8000",
      model: "nova-2",
      punctuate: true,
      interim_results: true,
      endpointing: 200,
      utterance_end_ms: 1000.
     smart_format: true
    }

**Comment by team-deepgram** [comment]
Thanks for asking your question about Deepgram! If you didn't already include it in your post, please be sure to add as much detail as possible so we can assist you efficiently, such as:
- The `request_id` if you have a question about your requests or transcription responses.
- The features you used or the full `api.deepgram.com` URL you sent your request to, including parameters.
- Any code snippets you can share.

**Comment by karanbangia** [comment]
@srisch

**Comment by jkroll-deepgram** [comment, marked as answer]
Hi @karanbangia, if you don't need the [interim results](https://developers.deepgram.com/docs/interim-results/), I recommend removing that parameter, so you only get final results (responses with `is_final: true`, not `is_final: false`).

I also suggest reading further into our [endpointing doc](https://developers.deepgram.com/docs/endpointing) and [end-of-speech detection](https://developers.deepgram.com/docs/understanding-end-of-speech-detection), to understand the difference between `speech_final` (end of speech/utterance) and `is_final` (finalized transcript). Depending on your goal within the conversational bot, you may want to use either or both in different cases. To get the fastest finalized transcript, look for `is_final`. To understand when a user has finished their utterance, use `speech_final`.

Another offering we have to shave off streaming latency is our [on-prem option](https://developers.deepgram.com/docs/on-prem-introduction), where you host Deepgram's models on your own servers, which can reduce latency by another 100-200 milliseconds or so. That's an enterprise offering with a yearly contract, let me know if that's of any interest to you as well.
